### ISSUE TITLE
1.7.1 broke sync

### ISSUE THREAD

**Issue by fr33z0n3r** [owncloud/android#940]
**UPDATE** - Temp workaround found:

**TL;DR:** They can't handle large storage sizes in their quota checks. Set a quota limit for your "Default Storage" value that just exceeds what you need. I put in 100GB.

--------------Original  ----------------------------------------------------------------------
I have just tested a copy of the 1.7.0 client and the 1.7.1 client.  It appears that there is some type of serious issue with 1.7.1 preventing "Refresh account" from working.

With my original install of OC Android 1.7.1 (upgraded via the Play Store over the past year) I could see a list of all my files, but I could not get a modified file to refresh its timestamp OR its content.

When I uninstalled 1.7.1 and installed FDroid 1.7.0 client, I lost my entire config of course, but upon adding in the correct server info, was able to sync all of my content.  FDroid OC 1.7.0 copy - https://f-droid.org/repo/com.owncloud.android_10700000.apk

I then uninstalled 1.7.0, and reinstalled 1.7.1 from the Play Store, and while it could connect to my server, it was again unable to sync my files, and the entire file listing is now empty.

Device:  Google Nexus 5
OS Version: 4.4.4 stock (not rooted)
OC Server: 6.0.2 on QNAP, unknown PHP version

**Comment by davivel** [comment]
Hi, @fr33z0n3r .

In the broken app, do you see a "progress circle" in the top bar when you start the, or when you pull down to refresh the list of files? Is it there forever, or does it disappear without the list being upgraded?

Besides, you should upgrade your server. The current version of OC 6 is 6.0.7. I cannot grant this is the case, but in the past we had refresh problems with a bug in the server preventing that updates of the list where propagated until the root folder.

Take a look to https://github.com/owncloud/core/wiki/Maintenance-and-Release-Schedule

**Comment by EtienneBruines** [comment]
I believe I might be experiencing the same issue. Before 1.7.1, synchronizing worked fine, but once I've downloaded 1.7.1, synchronizing (directories) is not working. 

The server is running `8.0.2`, and not OC 6. So my guess is that it is not server related. 

To answer your questions:
* Yes, the progress circle is there
* It stays there for quite some time, but then disappears
* The directory listing is not updated. As it does not list any directories, I don't know if it also applies to files. 

I can do this (which might give more insight as to where the problem is):
* Create a folder on mobile app
* View this folder online (web interface)
* Delete this folder online (web interface)
* Click "Refresh" on mobile app
* Still see the newly created folder (which I deleted in the web interface)
* Open the folder on mobile app
* Get an error message "Directory no longer exists" on mobile app (roughly translated)

(Android 5.1, custom rom)

**Comment by rahra** [comment]
I found out that the problem is the colon ':' within file names. If the files are renamed to something without ':', the download works.

**Comment by fr33z0n3r** [comment]
My issue has nothing to do with semi colons. When I tested the 1.7.1 app initially, the spinning cursor appeared for a while.  after I was forced to delete my config, it simply quickly flashes.

**Comment by fr33z0n3r** [comment]
can't edit my last post on mobile.

I meant "spinning progress icon".

Thanks to EtienneBruines for describing the issue well.  I don't have much time to test like he did until this weekend

**Comment by mhzawadi** [comment]
Hi, just want to add my voice to the list.

Last version of the app worked fine, but now I cant upload any files. When I try an upload the app shows in the notifications bar and a yellow up arrow on the folder, but it stays at 0%.

{"reqId":"11b13eeddbb0c8efdff311cd8c775751","remoteAddr":"212.159.117.222","app":"files_external","message":"Error: (401) Unauthorized\n","level":3,"time":"2015-04-18T13:50:35+00:00"}

**Comment by tobiasKaminsky** [comment]
@all can you test it with the demo owncloud server to eliminate your server as the cause of the problem?
demo.owncloud.org
user: test
password: test

**Comment by HFLaursen** [comment]
I have the same problem: android v1.7.1 sync no longer works; when using "refresh acount" sync circle runs for a few sec, then stops and notification "Synchronization failed, sync of (account@server) ...".

Uninstalling, clear cache and app data and delete working directory; and re-install results in no folders listed.

However, inst.upload still works and uploads new photos correctly to server. And this results in the Inst.upload folder is listed (as the only one).

The server still works with the pc (windows) and web clients.

WebDAV access works fine.

And the demo server (demo.owncloud.org) works fine from the android device, i.e. it lists folders as expected.

configuration: 

Android device: Samsung Galaxy S3 with ROM CM-11-20150412-nightly
Android owncloud client 1.7.1
owncloud server platform: QNAP NAS (linux)
owncloud server: 6.0.2 (latest official QNAP package, http://download.qnap.com/QPKG/owncloud_6.0.2.zip)

**Comment by fr33z0n3r** [comment]
I concur that the demo server works.  I'm hoping we can find an acceptable fix.

Many simply do not have the ability to perform server upgrades - like in the case of QNAP, whose setup is so delicate as to require vendor support.  The built-in upgrade support is very broken.

If someone can test any (or all) versions from 6.0.3 up to 6.0.7 to see if any of them fix this problem, that is the next step I think.  If none of them can fix this, then a major version upgrade would be required, and that is much less likely to happen on some of the installs.

**Comment by mhzawadi** [comment]
Hi,

The demo server is working for me also

I have oc v8.0.2 on a raspberry pi b+ 2

**Comment by fr33z0n3r** [comment]
OK, So I just managed to update my server to 6.0.7, which appears to be supported release AND....it still doesn't work.  I get "synchronization failed".

My SSL cert is self-signed, but unchanged in a year.  It is listed as valid using online SSL checker services (except that it is self signed).  I have had no issues using this specific mobile device and this specific OC server for a year.

Could you publish the changelog for mobile apps specifically?

**Comment by apiest** [comment]
Hi,

I've a 8.0.2 server {"installed":true,"maintenance":false,"version":"8.0.2.0","versionstring":"8.0.2","edition":""}

and files / directories in my folder but the app 1.7.1 is coming up empty.

**Comment by fr33z0n3r** [comment]
I have just found a copy of 1.7.0 apk with sha256 hash of e318a84f3d90d4b9c5f6c7bbe8b7257f2ad79bc054c4a89ca075244e43c6c873 by performing a google search. (https://www.google.com/search?channel=fs&q=owncloud+1.7.0+apk&ie=utf-8&oe=utf-8) 

Reviewing the cert inside shows the same values as seen in the Play Store version 1.7.1 .

Upon installation and server config of this 1.7.0 version, my folders and content are immediately made visible and syncable.

Please review the changes you have made in this version and try to help your users with identifying a fix.  As you can see, there are several users with newer major versions that are also being impacted.  

Since v6.0.7 has been tested and doesn't work with android 1.7.1, I would expect a server-side fix to be released for v6 regardless of how long that effort takes to identify it. Even if it goes past the published EOL for v6 OC server.

I'm hoping this is a simple issue. I will see about opening a OC server bug or getting assistance in troubleshooting from that side also.

**Comment by fr33z0n3r** [comment]
I'm not really an expert at java - I actually know VERY little.  But here is what I think is happening.

The long variable error occurs from the function mentioned in this log line:

E/ReadRemoteFolderOperation(13246): at com.owncloud.android.lib.common.network.WebdavEntry.<init>(WebdavEntry.java:140)

This leads me to:
https://github.com/owncloud/android-library/blob/master/src/com/owncloud/android/lib/common/network/WebdavEntry.java#L140

Which clearly indicates that this is looking at the users quotas available bytes
`
137            // {DAV:}quota-available-bytes
138            prop = propSet.get(DavPropertyName.create(PROPERTY_QUOTA_AVAILABLE_BYTES));
139            if (prop != null) {
140                mQuotaAvailableBytes = Long.parseLong((String) prop.getValue());
}`

My server has a "Default Limit" of Unlimited.  And my user account has a Quota of "Default".

All is fine and well here, right?

The value being returned at this line of code is "3.89279745229E+12", which isn't in decimal format.

The limit of a long variable in java is:
Maximum value is 9,223,372,036,854,775,807 (inclusive). (2^63 -1)

**You need to learn how to handle large numbers for your quota value check.**

http://stackoverflow.com/questions/2722122/java-parse-a-number-in-exponential-notation

Also, your demo server doesn't have this check.  Please consider that just because YOUR demo server works, it doesn't mean the server is leading to the issue.

**Comment by apiest** [comment]
@fr33z0n3r I can confirm the Default storage "hack" works.
The app is now displaying the folders again.

Thanx!

**Comment by HFLaursen** [comment]
@fr33z0n3r - I concur.  Setting server Default storage to e.g. 100 GB works.
Thanx much for this workaround.

_configuration:

Android device: Samsung Galaxy S3 with ROM CM-11-20150412-nightly
Android owncloud client 1.7.1
owncloud server platform: QNAP NAS (linux)
owncloud server: 6.0.2 (latest official QNAP package, http://download.qnap.com/QPKG/owncloud_6.0.2.zip)_

**Comment by masensio** [comment]
Hi @all,

I've seen that the workaround of @fr33z0n3r works for some of you. @EtienneBruines, @mhzawadi, @rahra, this solution is enough to solve your problem?

**Comment by EtienneBruines** [comment]
@masensio The workaround of @fr33z0n3r did not work for me. 

* Setting the "Default Storage" to 100 GB (or 40 GB for that matter)
* Setting the Storage Limit of my specific user account to "Default" (or unlimited, or 40GB, or 100GB)
* Removing the account at my phone
* Adding the account at my phone
* Synchronizing does not contain anything (except the InstantUpload folder my phone created). 

I shall attempt to view logs at my phone.

**Comment by EtienneBruines** [comment]
I did encounter the same exception as @fr33z0n3r documented:

```
E/ReadRemoteFolderOperation( 3798): Synchronized /: Unexpected exception
E/ReadRemoteFolderOperation( 3798): java.lang.NumberFormatException: Invalid long: "8.033313E+9"
E/ReadRemoteFolderOperation( 3798): 	at java.lang.Long.invalidLong(Long.java:124)
E/ReadRemoteFolderOperation( 3798): 	at java.lang.Long.parse(Long.java:363)
E/ReadRemoteFolderOperation( 3798): 	at java.lang.Long.parseLong(Long.java:353)
E/ReadRemoteFolderOperation( 3798): 	at java.lang.Long.parseLong(Long.java:321)
E/ReadRemoteFolderOperation( 3798): 	at com.owncloud.android.lib.common.network.WebdavEntry.<init>(WebdavEntry.java:140)
E/ReadRemoteFolderOperation( 3798): 	at com.owncloud.android.lib.resources.files.ReadRemoteFolderOperation.readData(ReadRemoteFolderOperation.java:143)
E/ReadRemoteFolderOperation( 3798): 	at com.owncloud.android.lib.resources.files.ReadRemoteFolderOperation.run(ReadRemoteFolderOperation.java:89)
E/ReadRemoteFolderOperation( 3798): 	at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:136)
E/ReadRemoteFolderOperation( 3798): 	at com.owncloud.android.operations.RefreshFolderOperation.fetchAndSyncRemoteFolder(RefreshFolderOperation.java:284)
E/ReadRemoteFolderOperation( 3798): 	at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:202)
E/ReadRemoteFolderOperation( 3798): 	at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:136)
E/ReadRemoteFolderOperation( 3798): 	at com.owncloud.android.syncadapter.FileSyncAdapter.synchronizeFolder(FileSyncAdapter.java:291)
E/ReadRemoteFolderOperation( 3798): 	at com.owncloud.android.syncadapter.FileSyncAdapter.onPerformSync(FileSyncAdapter.java:190)
E/ReadRemoteFolderOperation( 3798): 	at android.content.AbstractThreadedSyncAdapter$SyncThread.run(AbstractThreadedSyncAdapter.java:259)
```

**Comment by masensio** [comment]
Hi @EtienneBruines,

thanks for your log information. This is useful to find a solution for the problem.

**Comment by EtienneBruines** [comment]
@masensio I managed to "fix" the problem server-side. 

File `/var/www/owncloud/lib/private/connector/sabre/server.php`, then line 254, altering to: 
```php
$newProperties[200][$prop] = (int)$quotaInfo[1];
```

I do not know if this breaks any 32-bit stuff, but the android app doesn't seem to experience any problems now. 

I'm also not sure why line 248 is still 
```php
$newProperties[200][$prop] = (int)$quotaInfo[0];
```
My feeling tells me the crash should have happened (Android App) at line 134, where it parses the used bytes.

**Comment by davivel** [comment]
@fr33z0n3r , all you found is right, and really useful.

> You need to learn how to handle large numbers for your quota value check.

Not really. If we were expecting a number in exponential notation, we already knew how to handle it. Thanks anyway for the link.

@PVince81 , @DeepDiver1975 , should we handle exponetial format, or is this a bug? Is this generated by our server, or by SabreDAV? WebDAV specification does not seem clear to me about the format of the number there. What I could find in a first check is:

> Quota is counted in *Arabic numerals* expressed in strings. 

in https://tools.ietf.org/html/rfc4331#section-9 .

That suggest to me decimal notation only, but maybe the problem is the level of detail in the specification.


In any case, this will require to control correctly the exception, so there's work on our side.

**Comment by PVince81** [comment]
I'd say it would be best to not use exponential notation. I think the returned number need to be the exact value, not an approximation.

We should not modify server.php as this file is a copy of the one from Sabre (which is then removed in 8.1).
The proper fix would be to find out why the values of `getQuotaInfo()` are strings and convert them to integers in the Node class instead (or even in OC_Helper::getStorageInfo() or deeper)

I'll have a look later.

**Comment by PVince81** [comment]
I raised https://github.com/owncloud/core/issues/16240 to fix the exponential issue in core.

**Comment by masensio** [comment]
Hi @all,

the fix for this issue is now in our develop branch. It will be released in the next version (1.7.2).

Please, check if the issue is solved with the next version (coming soon). 
If the problem persists with 1.7.2 version, reopen the issue.

**Comment by andreicristianpetcu** [comment]
Thank you! @masensio

**Comment by toimelin** [comment]
updating to 1.7.2 did not fix syncing for me. However reinstalling the app to 1.7.2 did.

**Comment by andreicristianpetcu** [comment]
hmmm... not yet on F-Droid. I'm curious if updating will work for me or if I have to reinstall like you.

**Comment by andreicristianpetcu** [comment]
I tested it and it works. Unlike @toimelin I just did an update, I did not need to reinstall. (I used the one form PlayStore since it did not land in F-Droid yet)

**Comment by mhzawadi** [comment]
I still cant upload, I get an error in the owncloud logs and in the nginx access logs I see lot of upload requests.

{"reqId":"YU4CEgasn65Ynu2ss+Fq","remoteAddr":"212.159.117.222","app":"PHP","message":"Cannot modify header information - headers already sent by (output started at \/mnt\/data1\/var\/data\/websites\/3rd_party\/owncloud\/3rdparty\/sabre\/http\/lib\/Sapi.php:70) at \/mnt\/data1\/var\/data\/websites\/3rd_party\/owncloud\/3rdparty\/sabre\/http\/lib\/Sapi.php#58","level":3,"time":"2015-07-12T11:24:16+00:00"}

212.159.117.222 - matt [12/Jul/2015:11:24:19 +0000] "HEAD /remote.php/webdav/photos/HTC/IMAG0431%20(2).jpg HTTP/1.1" 200 0 "-" "Mozilla/5.0 (Android) ownCloud-android/1.7.2"

**Comment by andreicristianpetcu** [comment]
@mhzawadi For me uploading always worked. Might your issue be a different bug?

**Comment by mhzawadi** [comment]
@andreicristianpetcu it could be a different issue, but it all stopped working in 1.7.1.
Will have a look in core for a bug report, if not create one.

**Comment by mhzawadi** [comment]
I have had a look into this and it would seem to be an odd issue using nginx/php-fpm, there is an issue in the core https://github.com/owncloud/core/issues/17586 that a user has it all working but a user with nginx doesn't.

I have had a look at the sabre git repo and patched my Sapi.php file to match, but it still fails to upload. (https://github.com/fruux/sabre-http/blob/master/lib/Sapi.php)

So when thw android add sends a HEAD request, what is it looking for when it gets a reply?

mhorwood@SCI-00004 ~$ curl 'https://owncloud.horwood.biz/remote.php/webdav/photos/IMAG0434                                                  %20.jpg' --head --user ?:?
HTTP/1.1 200 OK
Server: nginx
Date: Mon, 13 Jul 2015 13:56:52 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding
X-Powered-By: PHP/5.4.41-0+deb7u1
Set-Cookie: ocet7lrrdrar=qehcs8ksvtouek83u8f1h3tk40; path=/; HttpOnly
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Pragma: no-cache
Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 's                                                  elf' 'unsafe-inline'; frame-src *; img-src *; font-src 'self' data:; media-src *; connect-                                                  src *
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Frame-Options: Sameorigin
X-Robots-Tag: none
Strict-Transport-Security: max-age=15768000; includeSubdomains

mhorwood@SCI-00004 ~$ curl -T Pictures/ed.jpg 'https://owncloud.horwood.biz/remote.php/webdav/photos/IMAG0434.jpg' --user ?:?

mhorwood@SCI-00004 ~$ curl 'https://owncloud.horwood.biz/remote.php/webdav/photos/IMAG0434.jpg' --head --user ?:?
HTTP/1.1 200 OK
Server: nginx
Date: Mon, 13 Jul 2015 14:03:08 GMT
Content-Type: image/jpeg
Content-Length: 28880
Connection: keep-alive
X-Powered-By: PHP/5.4.41-0+deb7u1
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Pragma: no-cache
Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src *; font-src 'self' data:; media-src *; connect-src *
X-XSS-Protection: 1; mode=block
X-Content-Type-Options: nosniff
X-Frame-Options: Sameorigin
X-Robots-Tag: none
Set-Cookie: ocet7lrrdrar=i12bcbh80g95fhsgv28cqk5ak3; path=/; secure; HttpOnly
Last-Modified: Mon, 13 Jul 2015 14:02:08 GMT
ETag: "0c1a57bb49506c5123de656a4f6c4457"
Content-Disposition: attachment
OC-ETag: "0c1a57bb49506c5123de656a4f6c4457"
Strict-Transport-Security: max-age=15768000; includeSubdomains

**Comment by fr33z0n3r** [comment]
I've just tested this and it seems to work.  Not sure if my test was valid or not.  Do I need to restart anything to be sure the quota is in effect?

Changed Owncloud server quota to Unlimited. (from 100GB)
Launched Owncloud Android client.
Verified that the correct "data" folders were present and that the GUI quickly loaded the content. (no loading icon)
Took photo on Android.
Confirmed photo was auto-uploaded by Owncloud Android client.
Confirmed photo was visible in the correct folder in the Owncloud Android client.

**Comment by enoch85** [comment]
I still see: 

```
Error	PHP	Cannot modify header information - headers already sent by (output started at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php:70) at /var/www/owncloud/3rdparty/sabre/http/lib/Sapi.php#58	2015-07-18T16:06:26+02:00
```

Server 8.1-3 
Client 1.7.2.

Shouldn't this be fixed?

**Comment by tobiasKaminsky** [comment]
@enoch The "cannot modify ..." error message is only mentioned by you and @mhzawadi. And he points out that it may be a problem with his nginx. 
Do you see the "real" error: 
E/ReadRemoteFolderOperation(13246): java.lang.NumberFormatException: Invalid long 
in your Logcat?
I think this is another error... Because the NumberFormat-Error is from within the app and the php error is from your owncloud server.
(But of course I can be wrong).

**Comment by enoch85** [comment]
@tobiasKaminsky Yes, I think you are wrong, for once. ;)

I run Apache + Ubuntu Server 14.04-2. I updated from server 8.0.4 to 8.1 (like everybody else) and haven't changed anything in particular.

Need some more info to debug?

**Comment by tobiasKaminsky** [comment]
@enoch85 Maybe you find something relevant in the logcat of your device?

**Comment by enoch85** [comment]
@tobiasKaminsky Logcat...? On my android device?

**Comment by tobiasKaminsky** [comment]
Yeah: Assuming that you have linux and adb installed: adb logcat | grep $(adb shell ps | grep com.owncloud | cut -c10-15)
There are also apps that can show logcat.
Then try a sync and see what the output in logcat is.

**Comment by mhzawadi** [comment]
Hi @tobiasKaminsky 

I have run the command you suggested, also run a test with http only to see what headers are sent.
File is on my owncloud, link below

https://owncloud.horwood.biz/index.php/s/tuYCjfUAET43izw - PW is apptest

**Comment by tobiasKaminsky** [comment]
For me this does not look like the error that is described here.
There seems no problem with the quota, but only with the upload.
@davivel can you look into this?

**Comment by enoch85** [comment]
@mhzawadi Thanks for the upload, seems my issue is the same.

**Comment by davivel** [comment]
@all 

A vague name in an issue does not mean that anything may go in. There is much more information in the first comment to have an idea about the real topic being dealt.

First comment:
> With my original install of OC Android 1.7.1 (upgraded via the Play Store over the past year) I could see a list of all my files, but I could not get a modified file to refresh its timestamp OR its content.

First report of "not really fixed" after the first close:
> Synchronizing does not contain anything (except the InstantUpload folder my phone created).

Both comments (and others) are complaining about an outdated or even empty (not refreshed) list of files. Nothing about uploads.

All the discussion following the first comment by @mhzawadi is a parallel topic, and should be in a different issue. Please, open a new one an copy the relevant information in it.

**Comment by enoch85** [comment]
@davivel New issue here: #1074
